tools/gen-perf-map.sh

```diff
@@ -5,7 +5,7 @@ if [ "$#" -lt 3 ]; then
     exit 1
 fi
 
-nm -DSC $1 | grep " [TVWu] " | awk '{$3=""; print$0}' > nm-output.txt
+nm -aSC $1 | grep " [TtVWu] " | awk '{$3=""; print$0}' > nm-output.txt
```

Comment:
Without "D", symbols of dynamic objects, such as dynamic libraries linked to binary,  will not be shown.  Can we use "aDSC" instead?

Reply:
Thanks for the reply. I think `-a` prints all symbols?

From the man page:

```
       -a
       --debug-syms
           Display all symbols, even debugger-only symbols; normally these are not listed.
```

I'd obivously be happy to specify both `-aD` but it looks in that case it falls back to just `-D`.

Are you seeing different behaviour on your version of `nm`?

I am using

```
inf@jupiter:~/build/iodlr/tools$ nm --version
GNU nm version 2.38-25.fc37
```

Reply:
OK, I took look at the man page of nm again. It says the following. That is why the "-aD" is the same as "-D".
> -D
>        --dynamic
>            Display the dynamic symbols rather than the normal symbols.  This
>            is only meaningful for dynamic objects, such as certain types of
>            shared libraries.

Then, I think we need to add an option to gen-perf-map.sh so that users can choose what they want to do.

Reply:
Ah I see what you mean now. That's indeed a bit lame that there is no flag to unconditionally show all. 

I am happy to add a flag. Anything specific you had in mind? 

Another thing I am wondering but this might be quite hacky, the remapping of the binary itself will always start at baseaddress 0? Could we use that to determine which flag is needed or will that break with something like fpic?

Reply:
I don't think that hacky way is a good idea. Because I don't think there is a good way to figure out if a file is a shared object or binary.

We can just add an option for "-aSC", and make "-DSC" as default.

Reply:
I have added a flag now. Let me know what you think or how you want it changed.

If ok then I will update the docs accordingly.